drivers/serial/uart_gecko.c

```diff
@@ -10,6 +10,7 @@
 #include <em_gpio.h>
 #include <em_cmu.h>
 #include <soc.h>
+#include <board.h>
```

Comment:
Because for efr32mg_sltb004a board the board.h includes the "soc_pinmap.h" which holds the UART pin settings.

The other silabs boards are setting the pins in the "soc_pinmap.h" header which is included by the "soc.h" header, and I think is not the best approach since pin configuration/muxing should be set at board level, not soc. What do you think?

Reply:
The mapping which location corresponds to which pin is a property of the soc. But the location that is actually chosen is part of the board. Or am I missing something here?

Reply:
Its a 50/50 case, in that we set some standard pin cfg in the soc headers for a number of SoCs.  The board ends up select a default or pre-cfg option in a lot of cases.  Eventually we will move pin cfg info into the device tree.  For now I'd keep with the same pattern that other SiLabs boards/SoC are using.

Reply:
Ok.

I'll update the PR #9042 as soon as this one gets merged.